app/Http/Controllers/Controller.php

```
@@ -98,58 +98,81 @@ public function forgot_password(Request $request)
+        $user = User::whereEmail($email)->first();
+        $invitation = TeamInvitation::query()
+            ->where('email', $email)
+            ->when($invitationUuid, fn ($query) => $query->where('uuid', $invitationUuid))
+            ->first();
+
+        if (! $user || ! $invitation || $invitation->hasExpired() || ! $this->invitationLinkMatchesToken($invitation, $token)) {
+            return null;
         }
 
-        return redirect()->route('login')->with('error', 'Invalid credentials.');
+        return Hash::check($password, $user->password) ? [$user, $invitation] : null;
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Facades\Hash' in method 'magicLinkCredentials'. (undefined)

(StaticAccess)